hackoregon/team-budget
# Travis runs Django app in Docker (and test script)

We now have a verified Docker container build, that can run in Travis.

This build will launch the Django app inside the Docker container, and will run the Django `tests.py` script successfully.

This will enable us to start wiring in unit tests that can be used to validate that a commit results in a healthy container.  If yes, the container can theoretically be deployed automatically to an AWS environment.  [The intention at first is to only deploy commits that come from **master** branch.]

# How to Verify (Travis)

You can verify that this code creates a successful build by examining this Travis build and its log:
https://travis-ci.org/hackoregon/team-budget/builds/211172973

# NOTE
Line 459 reports a failure:
`/home/travis/build.sh: line 62: ./budget_proj/bin/docker-push.sh: No such file or directory`

This is expected.  The `docker-push.sh` script is a stub reference that has not been enabled yet - no point pushing containers to AWS until (a) the Travis build results in something stable and (b) we have an AWS environment ready (being worked in the devops squad now).

# How to Verify (local)

Similar to the existing Docker container available in **master**, you should still be able to run this Docker container locally - I have tested it myself - once you have met the prerequisite:
- `git pull`, then `git checkout docker_and_travis_runs_app`
- copy the `/budget_proj/bin/env_template.sh` file to `env.sh` (we have abandoned the use of `project_config.py` for now in favour of the team's preferred approach)
- fill in the missing values - if you do not use the production AWS environment values, you'll have to manually perform the python migration and data import yourself to the new database you create.

Once you have met the prerequisite, simply launch `./budget_proj/bin/start-proj.sh` from the `team-budget` folder - that'll build the Docker container - and then you can browse to http://127.0.0.1:8000/.
